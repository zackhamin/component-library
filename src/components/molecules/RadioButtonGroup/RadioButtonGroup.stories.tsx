import { RadioButton } from 'Atoms/RadioButton';
import { IRadioButtonGroupProps, RadioButtonGroup } from './RadioButtonGroup';
import { Meta, Story } from '@storybook/react';

export default {
	title: 'molecules/RadioButtonGroup',
	component: RadioButtonGroup,
	argTypes: { onChange: { action: 'changed' } },
} as Meta;

const options = [
	{
		children: 'Test',
		value: 'Value 1',
		id: 'idTest',
	},
	{
		children: 'Test 2',
		value: 'Value 2',
		id: 'idTest2',
	},
];

const Template: Story<IRadioButtonGroupProps> = (args) => (
	<RadioButtonGroup {...args}>
		{options.map(({ id, value, children }) => (
			<RadioButton key={id} value={value} id={id}>
				{children}
			</RadioButton>
		))}
	</RadioButtonGroup>
);

const defaultProps = {
	name: 'Radio Name',
	defaultValue: 'defaultValue',
	isVertical: true,
};

export const Vertical = Template.bind({});
Vertical.args = {
	...defaultProps,
};

export const Horizontal = Template.bind({});
Horizontal.args = {
	...defaultProps,
	isVertical: false,
};
