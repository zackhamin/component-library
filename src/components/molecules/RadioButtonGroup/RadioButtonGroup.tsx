import classNames from 'classnames';
import { StyledOrientationWrapper } from './RadioButtonGroup.styles';
import React, { createContext } from 'react';
import { useEffect, useState } from 'react';

export interface IRadioButtonGroupProps {
	isVertical?: boolean;
	isRequired?: boolean;
	children: React.ReactNode;
	onChange: (e: React.ChangeEvent<HTMLInputElement>) => void;
	name: string;
	defaultValue: string;
}

export interface IRadioButtonContext {
	handleOnChange: (e: React.ChangeEvent<HTMLInputElement>) => void;
	name: string;
	isSelected: string;
	isRequired?: boolean;
}

export const RadioButtonGroupContext = createContext<
	IRadioButtonContext | undefined
>(undefined);

const RadioButtonGroup = ({
	children,
	isVertical,
	onChange,
	name,
	defaultValue,
	isRequired,
}: IRadioButtonGroupProps) => {
	const [isSelected, setIsSelected] = useState('');

	const handleOnChange = (e: React.ChangeEvent<HTMLInputElement>) => {
		setIsSelected(e.target.value);
		onChange(e);
	};
	useEffect(() => {
		setIsSelected(defaultValue);
	}, [defaultValue]);

	return (
		<RadioButtonGroupContext.Provider
			value={{ name, handleOnChange, isSelected, isRequired }}
		>
			<div role="radiogroup">
				<StyledOrientationWrapper
					className={classNames({
						vertical: isVertical,
					})}
				>
					{children}
				</StyledOrientationWrapper>
			</div>
		</RadioButtonGroupContext.Provider>
	);
};

export { RadioButtonGroup };
