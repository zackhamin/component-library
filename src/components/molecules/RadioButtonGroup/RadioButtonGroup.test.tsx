import { RadioButtonGroup } from '../../molecules/RadioButtonGroup/RadioButtonGroup';
import { render } from '@testing-library/react';

const defaultProps = {
	isVertical: false,
	name: 'Radio1',
	children: 'Option 1',
	value: 'opt1',
	isDisabled: false,
	id: 'opt1',
	defaultValue: 'opt1',
	onChange: jest.fn(),
};

describe('RadioButtonGroup', () => {
	it('renders horizontally correctly', () => {
		const { container } = render(<RadioButtonGroup {...defaultProps} />);
		expect(container).toMatchSnapshot();
	});
	it('renders vertically correctly', () => {
		const horizontalOptions = { ...defaultProps, isVertical: true };
		const { container } = render(<RadioButtonGroup {...horizontalOptions} />);
		expect(container).toMatchSnapshot();
	});
});
