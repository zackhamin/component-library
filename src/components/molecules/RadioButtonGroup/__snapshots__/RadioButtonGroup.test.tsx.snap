// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioButtonGroup renders horizontally correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: space-between;
  -ms-flex-wrap: space-between;
  flex-wrap: space-between;
  height: auto;
  width: 100%;
  padding: 0rem 1.6rem 2.4rem 1.6rem;
  box-sizing: border-box;
}

.c0.vertical {
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

<div>
  <div
    role="radiogroup"
  >
    <div
      class="c0"
    >
      Option 1
    </div>
  </div>
</div>
`;

exports[`RadioButtonGroup renders vertically correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: space-between;
  -ms-flex-wrap: space-between;
  flex-wrap: space-between;
  height: auto;
  width: 100%;
  padding: 0rem 1.6rem 2.4rem 1.6rem;
  box-sizing: border-box;
}

.c0.vertical {
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

<div>
  <div
    role="radiogroup"
  >
    <div
      class="c0 vertical"
    >
      Option 1
    </div>
  </div>
</div>
`;
